src/sys/socket/addr.rs

```diff
@@ -366,7 +366,7 @@ impl UnixAddr {
             .try_into()
             .unwrap();
 
-            #[cfg(bsd)]
+            #[cfg(any(bsd, target_os = "haiku"))]
```

Comment:
I think the cfg expression here should mirror the one at line 292.  Either change this one to:
```suggestion
            #[cfg(not(any(
                linux_android,
                target_os = "fuchsia",
                target_os = "illumos",
                target_os = "redox",
           )))]
```
or change the one at line 292 to:
```rust
#[cfg(not(any(bsd, target_os = "haiku")))]
```